### PR TITLE
Show room preview bar with maximised widgets

### PR DESCRIPTION
**Pull request by jryans**
This gives non-members a clear way to join a public room when maximised widgets are active.

While it is perhaps slightly visually odd to only have the preview bar under the main area, it's consistent with what we do when showing the regular timeline in a non-joined room (where opening the room info panel triggers a similar visual confusion), so I think it's better to follow consistency for now, and then re-evaluate the bar placement for all cases holistically.

Fixes https://github.com/vector-im/element-web/issues/21542

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Show room preview bar with maximised widgets ([\#8180](https://github.com/matrix-org/matrix-react-sdk/pull/8180)). Fixes vector-im/element-web#21542.

Preview: https://pr8180--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Review by t3chguy — CHANGES_REQUESTED** [review]
image

This misalignment looks *really* bad imo, should be rounded and leave the gap/gutter consistent all around

**Comment by turt2live** [comment]
I was sort of anticipating that it'd be in the right panel, personally. I don't have strong design opinions though.

**Comment by jryans** [comment]
> I was sort of anticipating that it'd be in the right panel, personally. I don't have strong design opinions though.

The current visual styling of the preview bar seems to assume a larger width than that, so that's why I went this way. I've been trying to reuse as much of what we already have where reasonable, but yes, there are various approaches, including:

* existing preview bar under the widget (the current approach here)
* existing preview bar under the widget and right panel
* new, smaller preview bar in the right panel (largest amount of new design and eng work)

File: src/components/structures/RoomView.tsx [hunk 1]
```diff
@@ -2104,9 +2104,11 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
         const showChatEffects = SettingsStore.getValue('showChatEffects');
 
         let mainSplitBody;
+        let mainSplitContentClassName;
```
Comment (t3chguy) [inline src/components/structures/RoomView.tsx:2107]:
```suggestion
        let mainSplitContentClassName: string;
```

File: src/components/views/right_panel/TimelineCard.tsx [hunk 2]
```diff
@@ -221,6 +221,9 @@ export default class TimelineCard extends React.Component<IProps, IState> {
 
         const isUploading = ContentMessages.sharedInstance().getCurrentUploads(this.props.composerRelation).length > 0;
 
+        const myMembership = this.props.room.getMyMembership();
+        const canSpeak = myMembership === "join";
```
Comment (t3chguy) [inline src/components/views/right_panel/TimelineCard.tsx:225]: This should be using something akin to https://github.com/matrix-org/matrix-react-sdk/blob/21d3cb08a9223cac99e1b6bce42090a828256135/src/components/views/rooms/MessageComposer.tsx#L368 to account also for permissions & tombstones.

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

**Comment by jryans** [comment]
> This misalignment looks _really_ bad imo, should be rounded and leave the gap/gutter consistent all around

Yes, it's not great... I have applied your suggestions, it now looks like:

image

The whole window view in the first comment has also been updated.

I have not done anything about the regular timeline case of preview bar plus right panel (which has a related affliction), since it's an existing issue and the solution seems a bit less obvious:

image

File: src/components/views/right_panel/TimelineCard.tsx (hunk 2 above)
Comment (jryans) [reply src/components/views/right_panel/TimelineCard.tsx:225]: Ah okay, I just copied from:

 https://github.com/matrix-org/matrix-react-sdk/blob/44776a1c5c81209abdad60b1766a3a963d0980a0/src/components/structures/RoomView.tsx#L1986-L1989

Should it be changed as well?

Comment (jryans) [reply src/components/views/right_panel/TimelineCard.tsx:225]: Will rename both my version and `RoomView`'s to `showComposer`, as that's what it really means (discussed [here](https://matrix.to/#/!bEWtlqtDwCLFIAKAcv:matrix.org/$9W3no-bx8A5SmFEqNYum3jA_AXt8lwU2isAtpQ9gbTg?via=matrix.org&via=element.io&via=vector.modular.im).

**Pull request merged**